Cython/Compiler/Nodes.py

```diff
@@ -5354,8 +5361,8 @@ def generate_execution_code(self, code):
         code.mark_pos(self.pos)
         self.expr.result_is_used = False  # hint that .result() may safely be left empty
         self.expr.generate_evaluation_code(code)
-        if not self.expr.is_temp and self.expr.result():
-            result = self.expr.result()
+        result = self.expr.result()
+        if not self.expr.is_temp and result:
```

Comment:
Could you keep the `is_temp` check before calling `result()` here? Just in case? Just split the `if` into two.

Reply:
Reverted to what it was before because I don't think the check in `result` is enforceable.